main.tf

```diff
@@ -88,7 +88,7 @@ locals {
 }
 
 data "null_data_source" "tags_as_list_of_maps" {
-  count = "${local.enabled ? length(keys(local.tags)) : 0}"
+  count = "${length(keys(var.additional_tag_map)) > 0 && local.enabled == true ? length(keys(local.tags)) : 0}"
```

Comment:
@jamie-wearsafe  thanks for the PR.
Output `tags_as_list_of_maps` is not related to the input variable `additional_tag_map`.
Output `tags_as_list_of_maps` is used to convert ALL tags into a different format (list of maps) that are required by some AWS resources (not all) regardless whether the input variable `additional_tag_map` provided or not (empty).
The input variable `additional_tag_map` is used to add additional tags to all resources in the format {key: xxxx, value: yyyy}.

`data "null_data_source" "tags_as_list_of_maps"` just converts ALL tags (generated and provided) to the same format {key: xxxx, value: yyyy} AND appends additional tags from `additional_tag_map` if provided.